source/ftrack_connect_pipeline_qt/ui/assembler/assembler.py

```diff
@@ -518,18 +525,19 @@ def _on_version_changed(self, widget, version_entity):
             matching_definitions = self.model.data(widget.index)[1]
             # Replace version ID for importer
             for definition in matching_definitions:
-                for context in definition['contexts']:
-                    for stage in context['stages']:
-                        if stage['name'] == 'context':
-                            for plugin in stage['plugins']:
-                                if 'options' in plugin:
-                                    options = plugin['options']
-                                    options['version_id'] = version_entity[
-                                        'id'
-                                    ]
-                                    options['version_number'] = version_entity[
-                                        'version'
-                                    ]
+                for context in definition.get_all(type=core_constants.CONTEXT):
+                    for stage in context.get_all(
+                        category=core_constants.STAGE
+                    ):
+                        for plugin in stage.get_all(
+                            category=core_constants.PLUGIN
+                        ):
+                            if 'options' in plugin:
+                                options = plugin['options']
+                                options['version_id'] = version_entity['id']
+                                options['version_number'] = version_entity[
+                                    'version'
+                                ]
```

Comment:
I might be wrong, but shouldn't this be the same and simplier?
```suggestion
                for plugin in definition.get_all(type=core_constants.CONTEXT, category=core_constants.PLUGIN):
                  if 'options' in plugin:
                      options = plugin['options']
                      options['version_id'] = version_entity['id']
                      options['version_number'] = version_entity[
                          'version'
                      ]
```